entity/index.js

```diff
@@ -325,27 +500,42 @@ EntityGenerator.prototype.files = function files() {
     insight.track('entity/relationships', this.relationships.length);
 
     var resourceDir = 'src/main/resources/';
-
+    
     this.template('src/main/java/package/domain/_Entity.java',
         'src/main/java/' + this.packageFolder + '/domain/' +    this.entityClass + '.java', this, {});
 
     this.template('src/main/java/package/repository/_EntityRepository.java',
         'src/main/java/' + this.packageFolder + '/repository/' +    this.entityClass + 'Repository.java', this, {});
-
-    this.template('src/main/java/package/web/rest/_EntityResource.java',
-        'src/main/java/' + this.packageFolder + '/web/rest/' +    this.entityClass + 'Resource.java', this, {});
-
+    
+    var fields = this.getInheritedFields(this.inheritanceFromClass, this.fields, this);
     if (this.databaseType == "sql") {
-        this.template(resourceDir + '/config/liquibase/changelog/_added_entity.xml',
-            resourceDir + 'config/liquibase/changelog/' + this.changelogDate + '_added_entity_' + this.entityClass + '.xml', this, {});
+        if(this.entityIsRootTable == false) {
+            this.baseTable = this.searchBaseTable(this.inheritanceFromClass, this);
+            this.template(resourceDir + '/config/liquibase/changelog/_added_inherited_entity.xml',
+                resourceDir + 'config/liquibase/changelog/' + this.changelogDate + '_added_inherited_entity_' + this.entityClass + '.xml', this, {});
+            
+            this.addChangelogToLiquibase(this.changelogDate + '_added_inherited_entity_' + this.entityClass);
+        } else {
+            if(this.inheritanceFromClass.length > 0) {
+                this.fields = fields;
+            }
+            this.template(resourceDir + '/config/liquibase/changelog/_added_entity.xml',
+                resourceDir + 'config/liquibase/changelog/' + this.changelogDate + '_added_entity_' + this.entityClass + '.xml', this, {});
 
-        this.addChangelogToLiquibase(this.changelogDate + '_added_entity_' + this.entityClass);
+            this.addChangelogToLiquibase(this.changelogDate + '_added_entity_' + this.entityClass);
+        }
+    }
+    
+    // after generating Liquibase we can merge fields from all inherited classes
+    if(this.inheritanceFromClass.length > 0) {
+        this.fields = fields;
+        this.fieldsTypes();
     }
+    this.template('src/main/java/package/web/rest/_EntityResource.java',
+            'src/main/java/' + this.packageFolder + '/web/rest/' +    this.entityClass + 'Resource.java', this, {});
 
     this.template('src/main/webapp/app/_entities.html',
         'src/main/webapp/scripts/app/entities/' +    this.entityInstance  + '/' + this.entityInstance + 's.html', this, {});
-    this.template('src/main/webapp/app/_entity-detail.html',
```

Comment:
That's my fault. I've got mess while merging and branching. This line should be there. I will fix this.